plugins/github/api.ml

```diff
@@ -451,13 +470,36 @@ let to_ci_refs refs =
   |> Ref_map.bindings
   |> List.map snd
 
-let ci_refs t repo =
+let now () =
+  Ptime.of_float_s (Unix.gettimeofday ()) |> function
+  | Some t -> t
+  | None -> Fmt.failwith "Failed to get current time"
+
+let to_ptime str =
+  Ptime.of_rfc3339 str |> function
+  | Ok (t, _, _) -> t
+  | Error _ -> Fmt.failwith "Datetime parsing error for: %s" str
```

Comment:
Any reason not to use `Ptime.pp_rfc3339_error` to show the original error here?

Reply:
Nope, will add, thanks :)